keras_rs/src/layers/retrieval/remove_accidental_hits.py

```diff
@@ -0,0 +1,83 @@
+import keras
+import numpy as np
+from keras import ops
+
+from keras_rs.src import types
+from keras_rs.src.api_export import keras_rs_export
+from keras_rs.src.utils import keras_utils
+
+SMALLEST_FLOAT = np.finfo(np.float32).smallest_normal / 100.0
+
+
+@keras_rs_export("keras_rs.layers.RemoveAccidentalHits")
+class RemoveAccidentalHits(keras.layers.Layer):
+    """Zeroes the logits of accidental negatives.
+
+    Zeroes the logits of negative candidates that have the same ID as the
+    positive candidate in that row.
+    """
+
+    def call(
+        self,
+        labels: types.Tensor,
+        logits: types.Tensor,
+        candidate_ids: types.Tensor,
+    ) -> types.Tensor:
+        """Zeroes selected logits.
+
+        For each row in the batch, zeroes the logits of negative candidates that
+        have the same ID as the positive candidate in that row.
+
+        Args:
+          labels: one-hot labels tensor, typically
+              `[batch_size, num_candidates]` but can have more dimensions or be
+              1d as `[num_candidates]`.
+          logits: logits tensor. Must have the same shape as `labels`.
+          candidate_ids: candidate identifiers tensor, can be `[num_candidates]`
+              or `[batch_size, num_candidates]` or have more dimensions as long
+              as they match the last dimensions of `labels`.
+
+        Returns:
+          logits: Modified logits.
+        """
+        # A more principled way is to implement
+        # `softmax_cross_entropy_with_logits` with a input mask. Here we
+        # approximate so by letting accidental hits have extremely small logits
+        # (SMALLEST_FLOAT) for ease-of-implementation.
+
+        labels_shape = ops.shape(labels)
+        labels_rank = len(labels_shape)
+        logits_shape = ops.shape(logits)
+        candidate_ids_shape = ops.shape(candidate_ids)
+        candidate_ids_rank = len(candidate_ids_shape)
+
+        if not keras_utils.check_shapes_compatible(labels_shape, logits_shape):
+            raise ValueError(
+                "The shape of `labels` and `logits` must match. Got "
```

Comment:
Nit: let's follow the same format everywhere: https://github.com/keras-team/keras-rs/blob/7553d191a1a25c39976b0494b439b66477a9c23d/keras_rs/src/utils/pairwise_loss_utils.py#L82. So, should be "Received: ..."